### DISCUSSION TITLE
"ValueError: feature_names mismatch" of use Predict(X_new), XGBoost in Python

### DISCUSSION THREAD

**Discussion by avryzhov** [pycaret/pycaret#2039] [Q&A] [answered]
Hello! Help me please.
I trained an XGBoost model, but when trying to make a prediction on new data, I get this error.

1.
#mob_inet,fix_inet,fix_ctv,fix_ictv,voice_mob,voice_fix,sms,csd,iot,mms,roaming,mg,mn,mts,conc,fix_op,vsr_roam,national_roam,mn_roam,voice_ap,voice_fee,period_service,one_time_service,dop_service,content,services_service,keo_sale,discount,only_inbound,sms_a2p,sms_gross,skoring,other_service,voice_mail,geo,ep_for_number,ep_for_line,one-time_fee_for_number,equipment rent,add_package
X_new = np.array([[0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,0,0,0,0,0,1,0,0,0]])

2. xgb_cl.predict(X_new)
-> run...

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-52-453dd134a682> in <module>
----> 1 xgb_cl.predict(X_new)
      2 #предсказан пятый класс

~/opt/anaconda3/lib/python3.8/site-packages/xgboost/sklearn.py in predict(self, data, output_margin, ntree_limit, validate_features, base_margin)
    968         if ntree_limit is None:
    969             ntree_limit = getattr(self, "best_ntree_limit", 0)
--> 970         class_probs = self.get_booster().predict(
    971             test_dmatrix,
    972             output_margin=output_margin,

~/opt/anaconda3/lib/python3.8/site-packages/xgboost/core.py in predict(self, data, output_margin, ntree_limit, pred_leaf, pred_contribs, approx_contribs, pred_interactions, validate_features, training)
   1483 
   1484         if validate_features:
-> 1485             self._validate_features(data)
   1486 
   1487         length = c_bst_ulong()

~/opt/anaconda3/lib/python3.8/site-packages/xgboost/core.py in _validate_features(self, data)
   2058                             ', '.join(str(s) for s in my_missing))
   2059 
-> 2060                 raise ValueError(msg.format(self.feature_names,
   2061                                             data.feature_names))
   2062 

ValueError: feature_names mismatch: ['mob_inet', 'fix_inet', 'fix_ctv', 'fix_ictv', 'voice_mob', 'voice_fix', 'sms', 'csd', 'iot', 'mms', 'roaming', 'mg', 'mn', 'mts', 'conc', 'fix_op', 'vsr_roam', 'national_roam', 'mn_roam', 'voice_ap', 'voice_fee', 'period_service', 'one_time_service', 'dop_service', 'content', 'services_service', 'keo_sale', 'discount', 'only_inbound', 'sms_a2p', 'sms_gross', 'skoring', 'other_service', 'voice_mail', 'geo', 'ep_for_number', 'ep_for_line', 'one-time_fee_for_number', 'equipment rent', 'add_package'] ['f0', 'f1', 'f2', 'f3', 'f4', 'f5', 'f6', 'f7', 'f8', 'f9', 'f10', 'f11', 'f12', 'f13', 'f14', 'f15', 'f16', 'f17', 'f18', 'f19', 'f20', 'f21', 'f22', 'f23', 'f24', 'f25', 'f26', 'f27', 'f28', 'f29', 'f30', 'f31', 'f32', 'f33', 'f34', 'f35', 'f36', 'f37', 'f38', 'f39']
expected iot, vsr_roam, discount, csd, one_time_service, voice_ap, period_service, other_service, geo, voice_fix, sms_a2p, fix_ctv, content, voice_mob, mn_roam, keo_sale, conc, voice_fee, mts, fix_ictv, dop_service, fix_op, mn, mg, fix_inet, national_roam, mms, only_inbound, voice_mail, sms_gross, add_package, sms, equipment rent, mob_inet, ep_for_line, ep_for_number, roaming, services_service, one-time_fee_for_number, skoring in input data
training data did not have the following fields: f25, f35, f39, f31, f11, f29, f22, f18, f12, f34, f27, f37, f20, f33, f4, f36, f3, f21, f14, f26, f38, f28, f9, f10, f6, f15, f2, f0, f30, f23, f7, f16, f5, f17, f13, f1, f8, f19, f32, f24

**Comment by moezali1** [comment, marked as answer]
@avryzhov Can you please open an issue on this? That may be a more appropriate place to answer this question.
